### PR TITLE
Bug 1835597: Add easier way to support selectors for multiple versions in OCS Tests

### PR DESCRIPTION
**Pull request by bipuladh**
- Add a way to write tests with support for multiple versions of selectors.
cc @ebondare @shyRozen

### CODE REVIEW

**Comment by bipuladh** [comment]
The motive behind this PR:
 - Declare selectors as objects for different versions.
 - Use environment variables to detect the OCP Version.
 - Based on the OCP version return the appropriate object that has the required selectors.

File: frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts [hunk 1]
```diff
@@ -2,47 +2,86 @@ import { $, ExpectedConditions as until, browser, $$ } from 'protractor';
...
+const OCP_44 = {
+  ocsOperator: $('a[data-test-operator-row="OpenShift Container Storage"]'),
+  ocsOperatorStatus: $('.co-clusterserviceversion-row__status'),
+  createLink: $('.pf-c-card__footer a'),
+  searchInputOperators: $('input[placeholder="Filter by name..."]'),
+};
+
+export const currentSelectors = (() => {
+  switch (VERSION) {
+    case 'OCP_44':
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts:79]: As discussed, can you please rename the version's name to `OCP_4.4`.

File: frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts [hunk 2]
```diff
@@ -2,47 +2,86 @@ import { $, ExpectedConditions as until, browser, $$ } from 'protractor';
...
+    // Node list fluctautes
+    await browser.wait(until.and(crudView.untilNoLoadersPresent));
+    await browser.wait(until.visibilityOf($('[aria-label="Node Table"] tbody tr')));
+    const rowCount = async () => $$('[aria-label="Node Table"] tbody tr').count();
+    await waitUntil(rowCount, 3, 5);
+  },
+
+  getStorageClusterLink: async () => {
+    const index = await getOperatorHubCardIndex('Storage Cluster');
+    const link = $(`article:nth-child(${index + 1}) a`);
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts:65]:
```suggestion
 return  $(`article:nth-child(${index + 1}) a`);
```

File: frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts [hunk 3]
```diff
@@ -67,54 +106,42 @@ export const goToWorkLoads = async () => {
...
-        selectedNodes.push(
-          await table
-            .$$('tbody a')
-            .get(index)
-            .getText(),
-        );
+          .getText();
+        await click(checkBox);
+        await browser.wait(until.elementToBeSelected(checkBox));
+        selectedNodes.push(nodeName);
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts:140]: Can you explain what are you checking here? Are you individually selecting nodes and not by select all nodes?

File: frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts [hunk 4]
```diff
@@ -130,78 +157,39 @@ export class InstallCluster {
     this.namespace = namespace;
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts:157]: Where have you used this property?

File: frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts [hunk 5]
```diff
@@ -130,78 +157,39 @@ export class InstallCluster {
...
-        .click();
-      await browser.wait(until.urlContains(`/${this.namespace}`), MINUTE);
+    await click(currentSelectors.ocsOperator);
+    // In fresh clusters APIs are not shown (Last seen in OCP 4.3)
+    try {
+      await browser.wait(until.visibilityOf(currentSelectors.createLink), 10 * SECOND);
+    } catch {
+      await refreshIfNotVisible(currentSelectors.createLink, 5);
     }
+    const storageClusterLink = await currentSelectors.getStorageClusterLink();
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts:185]: Can you point me to line where you have created `openshift-storage` namespace before installing ocs-operator

File: frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts (hunk 3 above)
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts:140]: On second though selecting all at once would be much easier. Thanks will look into it.

File: frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts (hunk 5 above)
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts:185]: One doesn't have to create the namespace to install ocs-operator. Operators have the capability to create names by themselves.

Comment (gnehapk) [reply frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts:185]: @umangachapagain Then why do we provide the namespace creation as a prerequisite step, while installing `ocs-operator`

**Comment by gnehapk** [comment]
@bipuladh Since you have removed the `ocp-tests` folder, the package.json also needs to be modified.

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/1-install/installFlow.scenario.ts [hunk 6]
```diff
@@ -66,101 +71,120 @@ describe('Test creation of Storage Cluster', () => {
...
+      await sendKeys(filterInput, STORAGE_CLASS_PATTERNS.RBD);
+      const rdbClass = await getDataFromRowAndCol(0, 1, podNameFilter);
+      expect(rdbClass.includes(STORAGE_CLASS_PATTERNS.RBD)).toBe(true);
+      await sendKeys(filterInput, STORAGE_CLASS_PATTERNS.FS);
+      const fsClass = await getDataFromRowAndCol(0, 1, podNameFilter);
+      expect(fsClass.includes(STORAGE_CLASS_PATTERNS.FS)).toBe(true);
+      await sendKeys(filterInput, STORAGE_CLASS_PATTERNS.NOOBAA);
+      const noobaaClass = await getDataFromRowAndCol(0, 1, podNameFilter);
+      expect(noobaaClass.includes(STORAGE_CLASS_PATTERNS.NOOBAA)).toBe(true);
+    });
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/1-install/installFlow.scenario.ts:188]: We should also add check for osd pods.

Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/integration-tests/tests/1-install/installFlow.scenario.ts:188]: osd pods cause flakiness.

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/1-install/installFlow.scenario.ts [hunk 7]
```diff
@@ -66,101 +72,128 @@ describe('Test creation of Storage Cluster', () => {
...
-    await sendKeys(filterInput, STORAGE_CLASS_PATTERNS.RBD);
-    const rdbClass = await getDataFromRowAndCol(0, 1, podNameFilter);
-    expect(rdbClass.includes(STORAGE_CLASS_PATTERNS.RBD)).toBe(true);
-    await sendKeys(filterInput, STORAGE_CLASS_PATTERNS.FS);
-    const fsClass = await getDataFromRowAndCol(0, 1, podNameFilter);
-    expect(fsClass.includes(STORAGE_CLASS_PATTERNS.FS)).toBe(true);
-    await sendKeys(filterInput, STORAGE_CLASS_PATTERNS.NOOBAA);
-    const noobaaClass = await getDataFromRowAndCol(0, 1, podNameFilter);
-    expect(noobaaClass.includes(STORAGE_CLASS_PATTERNS.NOOBAA)).toBe(true);
+    it('tests if 3 rook-ceph-osd-prepare-ocs-deviceset are running', () => {
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/1-install/installFlow.scenario.ts:178]:
```suggestion
    it('tests if 3 rook-ceph-osd-prepare-ocs-deviceset are completed', () => {
```

**Pull request merged**